experiments/multichain/multichain_client.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+from os import path, environ
+from sys import path as pythonpath
+import base64
+
+from gumby.experiments.dispersyclient import DispersyExperimentScriptClient, main
+
+from twisted.python.log import msg
+from twisted.internet import reactor
+
+pythonpath.append(path.abspath(path.join(path.dirname(__file__), '..', '..', '..', "./tribler")))
+
+from Tribler.dispersy.candidate import Candidate
+from Tribler.community.multichain.community import MultiChainCommunity, MultiChainCommunityCrawler, MultiChainScheduler
+
+
+class MultiChainClient(DispersyExperimentScriptClient):
+    """
+    Gumby client to start the MultiChain Community
+    """
+
+    def __init__(self, *args, **kwargs):
+        DispersyExperimentScriptClient.__init__(self, *args, **kwargs)
+        msg("Starting MultiChain client")
+        # Set the default MultiChainCommunity as community
+        self.community_class = MultiChainGumbyCommunity
+        self.scheduler = None
+        self.vars['public_key'] = base64.encodestring(self.my_member_key)
+
+    def registerCallbacks(self):
+        self.scenario_runner.register(self.introduce_candidates, 'introduce_candidates')
+        self.scenario_runner.register(self.set_multichain_type, 'set_multichain_type')
+        self.scenario_runner.register(self.request_signature, 'request_signature')
+        self.scenario_runner.register(self.request_crawl, 'request_crawl')
+        self.scenario_runner.register(self.increase_kbytes_sent, 'increase_kbytes_sent')
+        self.scenario_runner.register(self.increase_kbytes_received, 'increase_kbytes_received')
+        self.scenario_runner.register(self.schedule_block, 'schedule_block')
+        self.scenario_runner.register(self.close, 'close')
+
+    def set_multichain_type(self, multichain_type='MultiChainGumbyCommunity'):
+        """
+        Sets the community class of this gumby node to a special community.
+        """
+        msg("CommunityType: %s" % multichain_type)
+        if multichain_type == 'MultiChainDelayCommunity':
```

Comment:
It'd be nicer to `import Tribler.community.multichain.community as multichain_community` and do a hasattr()/getattr() instead of daisy-chaining all those if's.